src/components/Dropdown/Dropdown.test.js

```diff
@@ -0,0 +1,268 @@
+import React, { PureComponent } from 'react'
+import { render, fireEvent } from '@testing-library/react'
+import propTypes from 'prop-types'
+import {
+  Dropdown,
+  DropdownBody,
+  DropdownHeader,
+  DropdownFooter,
+  DropdownItem,
+  DropdownToggle,
+  DropdownContent,
+  DropdownContentControlled,
+} from '../index'
+
+describe('Dropdown Component', () => {
+  const CLASS_DROPDOWN_ACTIVE = 'mc-dropdown--active'
+  const CLASS_DROPDOWN_HTML_OPEN = 'mc-dropdown__html--open'
+
+  const toggleText = 'Click To Toggle'
+  const contentHeader = 'Some Header'
+  const contentBody = 'Some Content'
+  const contentFooter = 'Some Footer'
+
+  const dropdownIsVisible = (rootHtml, dropdownElem) => (
+    rootHtml.classList.contains(CLASS_DROPDOWN_HTML_OPEN) &&
+    dropdownElem.classList.contains(CLASS_DROPDOWN_ACTIVE)
+  )
+
+  describe('Basic DropdownContent', () => {
+    const renderSimpleDropdown = (
+      toggleText,
+      contentBody,
+      placement = 'bottom-start',
+    ) => (
+      <Dropdown placement={placement}>
+          <DropdownToggle><span>{toggleText}</span></DropdownToggle>
+          <DropdownContent>
+            <DropdownBody><span>{contentBody}</span></DropdownBody>
+          </DropdownContent>
+        </Dropdown>
+    )
+
+    const setup = () => {
```

Comment:
i tend to prefer that each test does it's own setup explicitly so that render and setup functions don't get more complex with arguments being passed to components as props, as well as having to move further up the test file to see what setup and render functions are actually doing.  this is pretty simple at the moment, and dropdowns don't have a tendency to change a whole lot, but for testing practices as a whole i'm not convinced this is a pattern we want to reproduce.  i'm not going to request it's changed, though.  we can just keep an eye on this

Reply:
yeah, it's always a balance.  My approach usually is to group tests together as much as possible using contexts (or describes), and then create a setup method for each context.  This helps to prevent crazy setup methods w/ lots of args.